cranelift/interpreter/src/value.rs

```diff
@@ -808,4 +819,97 @@ impl DataValueExt for DataValue {
     fn swap_bytes(self) -> ValueResult<Self> {
         unary_match!(swap_bytes(&self); [I16, I32, I64, I128])
     }
+
+    fn iter_lanes(&self, ty: Type) -> DataValueIterator {
+        DataValueIterator::new(self, ty)
+    }
+}
+
+/// Iterator for DataValue's
+pub struct DataValueIterator {
+    ty: Type,
+    v: SimdVec<DataValue>,
+    idx: usize,
+}
+
+impl DataValueIterator {
+    fn new(dv: &DataValue, ty: Type) -> Self {
+        match extractlanes(dv, ty) {
+            Ok(v) => return Self { ty, v, idx: 0 },
+            Err(_) => {
+                return Self {
+                    ty,
+                    v: SimdVec::new(),
+                    idx: 0,
+                }
+            }
+        }
+    }
+}
+
+impl Iterator for DataValueIterator {
+    type Item = DataValue;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.idx >= self.ty.lane_count() as usize {
+            return None;
+        }
+
+        let integer = self.v[self.idx].clone().into_int_signed().ok()?;
+        let dv = DataValue::from_integer(integer, self.ty.lane_type()).ok()?;
```

Comment:
I *think* we don't need the `into_int_signed` and the `from_integer`, If I'm reading this correctly we can just clone the value from the simd vec right?